openwisp_firmware_upgrader/api/serializers.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import serializers
+from swapper import load_model
+
+BatchUpgradeOperation = load_model('firmware_upgrader', 'BatchUpgradeOperation')
+Build = load_model('firmware_upgrader', 'Build')
+Category = load_model('firmware_upgrader', 'Category')
+FirmwareImage = load_model('firmware_upgrader', 'FirmwareImage')
+
+
+class BuildSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Build
+        fields = '__all__'
+
+
+class CategorySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Category
+        fields = '__all__'
+
+
+class BatchUpgradeOperationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = BatchUpgradeOperation
+        fields = '__all__'
```

Comment:
we need a nested serializer (readonly) in which we list the related operations (as in the admin), this is needed to monitor the progress of a mass upgrade

Reply:
Cool. Should I add a nested serializer for `Build` or do we want to have also fields like `completed`, `success_rate`, ...?

Reply:
Yes I think we should add a nested serializer and I also think it would be good to have `sucess_rate` and similar info in this view, it should be possible to accomplish it by adding specific methods to the serializer class.

Reply:
hey, I see that we have a nested serializer for the build object, that's good, but we also need a nested serializer which lists all the upgrade operations, is this clear?

Reply:
I see, it's clearer now :+1: